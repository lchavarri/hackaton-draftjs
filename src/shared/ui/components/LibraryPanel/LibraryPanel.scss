@import '../../assets/variables';
@import '../../assets/mixins';

.eidos-library {
  transition: margin-left $transition-normal;
  position: relative;
  width: $eidos-library-width;
  max-width: $eidos-library-width;
  z-index: 1;
  display: flex;
  flex-direction: column;

  &.collapsed {
    margin-left: -$eidos-library-width;

    & ~ * .eidos-library-sibling {
      padding-left: $eidos-library-width-collapsed;
    }

    & .eidos-library-toggle {
      transform: translateX($eidos-library-width);
    }
  }

  &-sibling {
    transition: padding-left $transition-normal;
  }

  &-toggle.button:not(.icon) {
    @include icon-button();
    height: $eidos-library-button-size;
    position: absolute;
    z-index: 20;
    top: rem(6px);
    left: $eidos-library-horizontal-padding;
    min-width: auto;
    border-radius: rem(4px);
    padding: rem(9px);
    text-transform: initial;
    font-size: rem(12px);

    .icon:not(.button):not(.dropdown) {
      height: rem(12px);
      width: rem(12px);
      margin: 0;
    }

    &.text .icon:not(.button):not(.dropdown) {
      height: rem(16px);
      width: rem(10px);
      margin-right: rem(16px);
    }
  }
}
