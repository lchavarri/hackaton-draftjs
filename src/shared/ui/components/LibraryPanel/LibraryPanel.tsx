import React, { useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes, faBars } from '@fortawesome/pro-light-svg-icons';

import './LibraryPanel.scss';

export default function LibraryPanel(props: any) {
  const [libraryCollapsed, setLibraryCollapsed] = useState(false);
  const title = props.title || 'Close Library Panel';
  const collapsedClass = libraryCollapsed ? 'collapsed' : '';

  const actionButton = libraryCollapsed ? (
    <button
      className="eidos-library-toggle ui button"
      onClick={() => setLibraryCollapsed(false)}
    >
      <FontAwesomeIcon icon={faBars} className="icon" aria-hidden="true" />
    </button>
  ) : (
    <button
      className="eidos-library-toggle ui button text"
      onClick={() => setLibraryCollapsed(true)}
    >
      <FontAwesomeIcon icon={faTimes} className="icon" aria-hidden="true" />{' '}
      {title}
    </button>
  );

  return (
    <>
      {actionButton}
      <aside className={`eidos-library ${collapsedClass}`}>
        {props.children}
      </aside>
    </>
  );
}
