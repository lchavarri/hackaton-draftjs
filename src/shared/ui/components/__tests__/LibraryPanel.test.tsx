import React from 'react';
import { shallow, mount, ReactWrapper } from 'enzyme';
import LibraryPanel from '../LibraryPanel/LibraryPanel';

describe('LibraryPanel component', () => {
  let component: ReactWrapper;
  let props: any;

  const defaultProps = {};
  const getComponent = () => <LibraryPanel {...props} />;
  const wrap = () => shallow(getComponent());
  const wrapDeep = () => mount(getComponent());

  beforeEach(() => {
    props = defaultProps;
    component = wrap();
  });

  it('should match component snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
