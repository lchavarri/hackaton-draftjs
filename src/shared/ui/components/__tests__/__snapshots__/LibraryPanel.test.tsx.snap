// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibraryPanel component should match component snapshot 1`] = `
<Fragment>
  <button
    className="eidos-library-toggle ui button text"
    onClick={[Function]}
  >
    <mockConstructor
      aria-hidden="true"
      className="icon"
      icon={[MockFunction]}
    />
     
    Close Library Panel
  </button>
  <aside
    className="eidos-library "
  />
</Fragment>
`;
